sdk/src/main/kotlin/io/rover/sdk/data/events/AnalyticsService.kt

```
@@ -67,6 +69,7 @@ internal class AnalyticsService(
             is RoverEvent.ScreenDismissed -> "Screen Dismissed" to eventInformation.toFlat()
             is RoverEvent.ScreenViewed -> "Screen Viewed" to eventInformation.toFlat()
             is RoverEvent.BlockTapped -> "Block Tapped" to eventInformation.toFlat()
+            else -> throw RuntimeException("Event not supported")
```

What is going to eat this error? We definitely don't want to be crashing when some other event than the above set is emitted.

---

Added a catch on the calling function to log exceptions thrown with the message